src/components/Tabs/components/Tab/components/DuplicateModal/DuplicateModal.vue

```diff
@@ -0,0 +1,129 @@
+<template lang="pug">
+Modal(
+  :instant="true",
+  :open="open",
+  :primaryAction="primaryAction",
+  :secondaryActions="secondaryActions",
+  @close="emits('close')",
+)
+  template(#title) {{ i18n.translate('Polaris.Tabs.DuplicateModal.title') }}
+  ModalSection
+    Form(@submit="handlePrimaryAction")
+      FormLayout
+        div(ref="container")
+          TextField(
+            v-model="model",
+            autoComplete="off",
+            :label="i18n.translate('Polaris.Tabs.DuplicateModal.label')",
+            :helpText="helpText",
+            :maxLength="MAX_VIEW_NAME_LENGTH",
+            :showCharacterCount="true",
+            :error="errorMessage",
+            @change="handleChange",
+          )
+</template>
+
+<script setup lang="ts">
+import { ref, computed, watch } from 'vue';
+import useI18n from '@/use/useI18n';
+import { focusFirstFocusableNode } from '@/utilities/focus';
+import { Modal, Form, ModalSection, FormLayout, TextField } from '@/components';
+
+export interface DuplicateModalProps {
+  open: boolean;
+  isModalLoading?: boolean;
+  name: string;
+  helpText?: string;
+  viewNames?: string[];
+}
+
+const MAX_VIEW_NAME_LENGTH = 40;
+
+const props = defineProps<DuplicateModalProps>();
+
+const emits = defineEmits<{
+  (e: 'close'): void;
+  (e: 'click-primary-action', value: string): Promise<boolean>;
```

Comment:
Why it be Promise?

Reply:
Because onClickPrimaryAction from the parent component is the Promise function